src/main/java/net/dries007/tfc/common/items/TFCBucketItem.java

```diff
@@ -0,0 +1,215 @@
+
+            for (int i = 0; i < 8; ++i)
+            {
+                level.addParticle(ParticleTypes.LARGE_SMOKE, pos.getX() + Math.random(), pos.getY() + Math.random(), pos.getZ() + Math.random(), 0.0D, 0.0D, 0.0D);
+            }
+            return true;
+        }
+        else if (block instanceof LiquidBlockContainer container && container.canPlaceLiquid(level, pos, state, content))
+        {
+            container.placeLiquid(level, pos, state, content.defaultFluidState());
+            playEmptySound(content, player, level, pos);
+            return true;
+        }
+        else
+        {
+            if (!level.isClientSide && replaceable && !material.isLiquid())
+            {
+                level.destroyBlock(pos, true);
+            }
+
+            if (!level.setBlock(pos, content.defaultFluidState().createLegacyBlock(), 11) && !state.getFluidState().isSource())
+            {
+                return false;
+            }
+            else
+            {
+                Helpers.playSound(level, pos, SoundEvents.BUCKET_EMPTY);
+                this.playEmptySound(content, player, level, pos);
+                return true;
+            }
+        }
+    }
+
+    protected void playEmptySound(Fluid content, @Nullable Player player, LevelAccessor level, BlockPos pos)
+    {
+        SoundEvent sound = content.getAttributes().getEmptySound();
+        if (sound == null)
+        {
+            sound = Helpers.isFluid(content, FluidTags.LAVA) ? SoundEvents.BUCKET_EMPTY_LAVA : SoundEvents.BUCKET_EMPTY;
+        }
+        level.playSound(player, pos, sound, SoundSource.BLOCKS, 1.0F, 1.0F);
+        level.gameEvent(player, GameEvent.FLUID_PLACE, pos);
+    }
+
+    protected InteractionResultHolder<ItemStack> afterFillFailed(IFluidHandler handler, Level level, Player player, ItemStack stack, InteractionHand hand)
+    {
+        return InteractionResultHolder.pass(stack);
+    }
+
+    protected InteractionResultHolder<ItemStack> afterEmptyFailed(IFluidHandler handler, Level level, Player player, ItemStack stack, InteractionHand hand)
+    {
+        return InteractionResultHolder.pass(stack);
+    }
+
+    public int getCapacity()
+    {
+        return capacity.get();
+    }
+
+    abstract TagKey<Fluid> getWhitelistTag();
+
+    @Nullable
+    @Override
+    public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt)
+    {
+        return new ItemStackFluidHandler(stack, getWhitelistTag(), getCapacity());
+    }
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> tooltip, TooltipFlag flag)
+    {
+        FluidStack fluid = stack.getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY).map(cap -> cap.getFluidInTank(0)).orElse(FluidStack.EMPTY);
+        if (!fluid.isEmpty()) Helpers.addFluidStackTooltipInfo(fluid, tooltip);
+    }
```

Comment:
Buckets shouldn't have a tooltip, they should be in the title, can we not modify that rather than this? Do the ole fluid + " " + bucket. with components ovbiously.